code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -572,32 +564,46 @@ GLOBAL_LIST_INIT(gas_id_to_canister, init_gas_id_to_canister())
 		else
 			shielding_powered = FALSE
 
+///return the icon_state component for the canister's indicator light based on its current pressure reading
+/obj/machinery/portable_atmospherics/canister/proc/get_pressure_state(air_pressure)
+	switch(air_pressure)
+		if((40 * ONE_ATMOSPHERE) to INFINITY)
+			return 3
+		if((10 * ONE_ATMOSPHERE) to (40 * ONE_ATMOSPHERE))
+			return 2
+		if((5 * ONE_ATMOSPHERE) to (10 * ONE_ATMOSPHERE))
+			return 1
+		if((10) to (5 * ONE_ATMOSPHERE))
+			return 0
+		else
+			return -1
```

Comment:
although, this could just return the icon state to use, and then you could wrap that for the light. might be a bit easier to track.

Reply:
Didnt think of that. Ill make it return the icon state directly

Reply:
Done